crates/ruff_linter/src/rules/pyupgrade/snapshots/ruff_linter__rules__pyupgrade__tests__UP050.py.snap

```diff
@@ -0,0 +1,322 @@
+24 24 | class C[T](Generic[_B], str): ...
+25    |-class C[T](int, Generic[_C]): ...
+   25 |+class C[T, _C: (str, bytes)](int): ...
+26 26 | class C[T](bytes, Generic[_D], bool): ...
+27 27 | class C[T](Generic[_E], list[_E]): ...
+28 28 | class C[T](list[_F], Generic[_F]): ...
+
+UP050.py:26:19: UP050 [*] Class with type parameter list inherits from `Generic`
+   |
+24 | class C[T](Generic[_B], str): ...
+25 | class C[T](int, Generic[_C]): ...
+26 | class C[T](bytes, Generic[_D], bool): ...
+   |                   ^^^^^^^^^^^ UP050
+27 | class C[T](Generic[_E], list[_E]): ...
+28 | class C[T](list[_F], Generic[_F]): ...
+   |
+   = help: Convert to new-style
+
+ℹ Unsafe fix
+23 23 | class C[T](Generic[_A]): ...
+24 24 | class C[T](Generic[_B], str): ...
+25 25 | class C[T](int, Generic[_C]): ...
+26    |-class C[T](bytes, Generic[_D], bool): ...
+   26 |+class C[T, _D](bytes, bool): ...
+27 27 | class C[T](Generic[_E], list[_E]): ...
+28 28 | class C[T](list[_F], Generic[_F]): ...
+29 29 | 
+
+UP050.py:27:12: UP050 [*] Class with type parameter list inherits from `Generic`
+   |
+25 | class C[T](int, Generic[_C]): ...
+26 | class C[T](bytes, Generic[_D], bool): ...
+27 | class C[T](Generic[_E], list[_E]): ...
+   |            ^^^^^^^^^^^ UP050
+28 | class C[T](list[_F], Generic[_F]): ...
+   |
+   = help: Convert to new-style
+
+ℹ Unsafe fix
+24 24 | class C[T](Generic[_B], str): ...
+25 25 | class C[T](int, Generic[_C]): ...
+26 26 | class C[T](bytes, Generic[_D], bool): ...
+27    |-class C[T](Generic[_E], list[_E]): ...
+   27 |+class C[T, _E: int](list[_E]): ...
+28 28 | class C[T](list[_F], Generic[_F]): ...
+29 29 | 
+30 30 | class C[*Ts](Generic[_As]): ...
+
+UP050.py:28:22: UP050 [*] Class with type parameter list inherits from `Generic`
+   |
+26 | class C[T](bytes, Generic[_D], bool): ...
+27 | class C[T](Generic[_E], list[_E]): ...
+28 | class C[T](list[_F], Generic[_F]): ...
+   |                      ^^^^^^^^^^^ UP050
+29 |
+30 | class C[*Ts](Generic[_As]): ...
+   |
+   = help: Convert to new-style
+
+ℹ Unsafe fix
+25 25 | class C[T](int, Generic[_C]): ...
+26 26 | class C[T](bytes, Generic[_D], bool): ...
+27 27 | class C[T](Generic[_E], list[_E]): ...
+28    |-class C[T](list[_F], Generic[_F]): ...
+   28 |+class C[T, _F: (str, bytes)](list[_F]): ...
+29 29 | 
+30 30 | class C[*Ts](Generic[_As]): ...
+31 31 | class C[*Ts](Generic[_Bs], tuple[*Bs]): ...
+
+UP050.py:30:14: UP050 [*] Class with type parameter list inherits from `Generic`
+   |
+28 | class C[T](list[_F], Generic[_F]): ...
+29 |
+30 | class C[*Ts](Generic[_As]): ...
+   |              ^^^^^^^^^^^^ UP050
+31 | class C[*Ts](Generic[_Bs], tuple[*Bs]): ...
+32 | class C[*Ts](Callable[[*_Cs], tuple[*Ts]], Generic[_Cs]): ...
+   |
+   = help: Convert to new-style
+
+ℹ Unsafe fix
+27 27 | class C[T](Generic[_E], list[_E]): ...
+28 28 | class C[T](list[_F], Generic[_F]): ...
+29 29 | 
+30    |-class C[*Ts](Generic[_As]): ...
+   30 |+class C[*Ts, *_As]: ...
```

Comment:
The correct way to make a class generic over a `TypeVarTuple` using the old-style syntax is to do either this:

```py
class C(Generic[*_As]): ...
```

or this:

```py
class C(Generic[Unpack[_As]]): ...
```

(We don't handle `Unpack` yet in our PEP-695 rules -- that's also an oustanding task in https://github.com/astral-sh/ruff/issues/15642)

I'm not sure we should try to autofix classes that are invalidly generic over `TypeVarTuple`s like this.